documentation/docs/customjenkins.md

```diff
@@ -0,0 +1,122 @@
+
+   <p align="center">
+   ![Add Library](images/JenkinsConfigSystemLibrary-Add.png "Add Library")
+   </p>
+
+1. For **Library Name** enter the library name `piper-lib-os`
+
+1. For **Default Version** enter the branch or tag you want to consume (e.g. `master` or `v0.1`)
+
+1. For **Retrieval Method** choose **Modern SCM**
+
+1. For **Source Code Management** choose **Git**
+
+1. For **Project Repository** enter the GitHub URL of the project Piper shared library `https://github.com/SAP/jenkins-library`
+
+   <p align="center">
+   ![Library Setup](images/JenkinsConfigSystemLibrary-Edit.png "Library Setup")
+   </p>
+
+1. Save changes
+
+**Result:** The library is available as `piper-lib-os` and can be used in any pipeline by adding the following line to its `Jenkinsfile`:
+
+```groovy
+@Library('piper-lib-os') _
+```
+
+When the pipeline is launched, Jenkins downloads the corresponding library as source and compiles it before the pipeline is processed.
+
+
+## User Permission Issue
+
+Your native Jenkins installation defines the user `jenkins` as service user. If the user doesn't exists it will be created. The user id will be the next free number determined by `/etc/passwd` - probably starting from `100`.
+In contrast, the official [Jenkins Docker image][jenkins-docker-image] defines the user `jenkins` with the userid `1000` as service user inside the container. 
+So, the service user id of your native Jenkins server most likely differs from the user id of the official Jenkins Docker image.
+
+This could have impacts. 
+
+Project "Piper" is running many pipeline steps as Docker images. If a Docker container is created, the Jenkins Docker plugin passes the Jenkins user and group id as process owner into the Docker container.
+Binding a folder from the host machine into the container - used to exchange files between steps - results in file permission issues if the user used inside the container doesn't have rights to the folder on the host machine or vice versa.
+
+You wont face this issue with images of the project "Piper" but, some 3rd party docker images follow this convention and expect to be executed under userid `1000`, like [node.js][dockerhub-node] which is used by a set of additional steps. 
+
+If you run into such [user permission issue][piper-issue-781] you have following options
+
+1. Use id `1000` - change the id of your Jenkins service user to `1000`.
+
+1. [Create your own images][docker-getstarted] - derive from the Docker image in question and solve the permission issues by removing the file system restrictions. Adjust the configuration accordingly, e.g.adjust the `npmExecute` step of your project's YAML:
+
+   ```
+     npmExecute:
+       dockerImage: 'my-node:8-stretch'
+   ```
+
+1. Setup a namespace - the user permission [issue 781][piper-issue-781] of the piper repository described how to setup a Linux kernel user namespace to circumvent the mismatch of user ids. This solution is experimental and should be well-considered.
```

Comment:
```suggestion
- Set up a namespace. The user permission [issue 781][piper-issue-781] of the project "Piper" repository describes how to set up a Linux kernel user namespace to prevent the mismatch of user IDs. **Note:** This solution is experimental and should be well-considered.
```